### PR TITLE
feat: cookie authentication

### LINKED ISSUE #241: Cookie based auth
See https://github.com/atomicdata-dev/atomic-data-docs/pull/138/files#

- Create an Authentication Resource with a suiting expiration date (e.g. 1 hrs, for testing purposes even shorter)
- Sign an authentication resource (re-use the logic found in websockets.ts) on app initialization 
- Set it as a cookie, make sure it expires on the same moment as the signed authentication resource. Set the `Secure` attribute.
- Make sure it works by creating a new Drive, uploading an Image and checking if you can view it

Of course, this also requires the back-end implementation

### PR DESCRIPTION
**Pull request by rescribet**
- [x] Closes #241 
- [x] Make sure E2E tests of most recent JS bundle passes in [atomic-data-rust PR](https://github.com/atomicdata-dev/atomic-data-browser/pull/246)
- [x] Update CHANGELOG.md, link to issue number
- [x] Resolve open PR review comments

### CODE REVIEW

File: data-browser/src/helpers/setCookie.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const ONE_HOUR = 60 * 60 * 1000;
+
+export const setCookie = (name: string, value: string) => {
+  const expiry = new Date(Date.now() + ONE_HOUR).toISOString();
```
Comment (joepio) [inline data-browser/src/helpers/setCookie.ts:4]: What happens if the expires date is passed? A new cookie should probably be set somewhere (auth error handler?)

**Comment by joepio** [comment]
Custom checklist for PR (for some reason the template is missing in this PR):

- [ ] Include updated JS assets (`pnpm build-server`) for the client-side Cookie implementation
- [ ] Update Contribute.md, link to issue number

**Comment by rescribet** [comment]
You probably mean the changelog? That already has a new entry with the ticket number.

**Comment by rescribet** [comment]
From what I can tell the test should succeed, but this PR isn't tested against that version of the server.

**Comment by joepio** [comment]
> From what I can tell the test should succeed, but this PR isn't tested against that version of the server.

Yes we can ignore this E2E test, as this uses the master docker build. The rust PR E2E must succeed, though.

File: data-browser/src/helpers/setCookie.ts (hunk 1 above)
Comment (rescribet) [reply data-browser/src/helpers/setCookie.ts]: The current version is updated whenever the agent change function is called (setAgentIfChanged).

**Pull request merged**